doc/contributing/docs/api.rst

```diff
@@ -0,0 +1,170 @@
+
+See :ref:`module function example <contributing-api-docs_function-example>`,
+:ref:`class method example <contributing-api-docs_class-example>`.
+
+Data
+^^^^
+
+*   :ref:`Since which Tarantool version <contributing-docs-api_version>`
+*   :ref:`General description <contributing-api-docs_general-description>`
+*   Return type
+*   Example
+
+See :ref:`class data example <contributing-api-docs_class-example>`.
+
+..  _documenting_parameters:
+
+Function and class parameters
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+*   :ref:`Since which Tarantool version <contributing-docs-api_version>`
+    (if added later)
+*   :ref:`General description <contributing-api-docs_general-description>`
+*   Type
+*   Default value (if optional), possible values
+
+If the parameter is optional, make sure it is enclosed in square brackets
+in the function declaration (in the "heading").
+
+In the "Possible errors" section of the function or class method,
+consider explaining what happens if the parameter hasn't been defined or has the wrong value.
+
+..  _documenting_confvals:
+
+Configuration parameters
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+Configuration parameters are not to be confused with class and method parameters.
+Configuration parameters are passed to Tarantool via the command line or in an initialization file.
+You can find a list of Tarantool configuration parameters in the :doc:`configuration reference </reference/configuration/index>`.
+
+*   :ref:`Since which Tarantool version <contributing-docs-api_version>`
+*   :ref:`General description <contributing-api-docs_general-description>`
+*   Type
+*   Corresponding environment variable (if applicable)
+*   Default value
+*   Possible values
+*   Dynamic (yes or no)
+
+See :ref:`configuration parameter example <contributing-api-docs_confval-example>`.
+
+Examples and templates
+----------------------
+
+..  _contributing-api-docs_function-example:
+
+Module functions
+^^^^^^^^^^^^^^^^
+
+We use the Sphinx directives ``.. module::``
+and ``.. function::`` to describe functions of Tarantool modules:
+
+..  literalinclude:: ./_includes/function_template.rst
+    :language: rst
+
+The resulting output looks like this:
+
+..  include:: ./_includes/function_template.rst
+
+
+..  _contributing-api-docs_class-example:
+
+Class methods and data
+^^^^^^^^^^^^^^^^^^^^^^
+
+Methods are described similarly to functions, but the ``.. class::``
+directive, unlike ``.. module::``, requires nesting.
+
+As for data, it's enough to write the description, the return type, and an example.
+
+Here is the example documentation describing
+the method and data of the ``index_object`` class:
+
+..  literalinclude:: ./_includes/class_template.rst
+    :language: rst
+
+And the resulting output looks like this:
+
+..  include:: ./_includes/class_template.rst
+
+..  _contributing-api-docs_confval-example:
+
+Configuration parameters
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+Example:
+
+..  literalinclude:: ./_includes/confval_template.rst
```

Comment:
Maybe we can give here another example? Of a parameter with a number of possible values -- to illustrate this checklist item.

Reply:
https://www.tarantool.io/en/doc/latest/reference/configuration/#confval-election_mode
This is the only one with "possible values" provided in a list. It has a huge description, so I wouldn't include it in this specific article.
I think that the possible values may, but need not, be a list.